src/dcegm/egm.py

```diff
@@ -0,0 +1,458 @@
+"""Auxiliary functions for the EGM algorithm."""
+from typing import Callable
+from typing import Tuple
+
+import numpy as np
+from dcegm.interpolate import interpolate_policy
+from dcegm.interpolate import interpolate_value
+
+
+def compute_optimal_policy_and_value(
+    marginal_utilities_exog_process,
+    max_value_func_exog_process,
+    discount_factor,
+    interest_rate: float,
+    choice: int,
+    trans_vec_state: np.ndarray,
+    savings_grid: np.ndarray,
+    compute_inverse_marginal_utility: Callable,
+    compute_value: Callable,
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Runs the Endogenous-Grid-Method Algorithm (EGM step).
```

Comment:
Need a better function description. The docstring probably comes from some previous copy-pasting.

Reply:
Changed descriptioN!